velox/functions/prestosql/types/TimestampWithTimeZoneType.h

```diff
@@ -125,4 +124,18 @@ class TimestampWithTimeZoneTypeFactories : public CustomTypeFactories {
 
 void registerTimestampWithTimeZoneType();
 
+static constexpr int32_t kTimezoneMask = 0xFFF;
+static constexpr int32_t kMillisShift = 12;
+
+inline int64_t unpackMillisUtc(int64_t dateTimeWithTimeZone) {
+  return dateTimeWithTimeZone >> kMillisShift;
+}
+
+inline int16_t unpackZoneKeyId(int64_t dateTimeWithTimeZone) {
```

Comment:
We should define `using TimeZoneKey = int16_t;` to make this clear and reuse it for tests.

Reply:
Refactored. Thank you @majetideepak deepak.